src/oop/programs_with_multiple_classes/hard/q4/MainTest.java

```diff
@@ -0,0 +1,150 @@
+package oop.programs_with_multiple_classes.hard.q4;
+
+import global.ObjectTest;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import java.time.Year;
+import java.util.stream.Stream;
+
+import static global.tools.CustomAssertions._assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+public class MainTest {
+    // Java
+
+    private ObjectTest book;
+    private ObjectTest bookcase;
+
+    private final String bookLc = "book";
+    private final String bookUc = "Book";
+    private final String bookcaseLc = "bookcase";
+    private final String bookcaseUc = "Bookcase";
+    private final String year = "year";
+    private final String type = "type";
+    private final String sizeCapacity = "sizeCapacity";
+    private final String b1 = "b1";
+
+    private final String determineTime = "determineTime";
+
+
+    @BeforeEach
+    public void Setup() {
+        String bookClassString = "oop.programs_with_multiple_classes.hard.q4." + bookUc;
+        String bookcaseClassString = "oop.programs_with_multiple_classes.hard.q4." + bookcaseUc;
+        book = new ObjectTest(bookClassString);
+        bookcase = new ObjectTest(bookcaseClassString);
+    }
+
+    /**
+     * Book tests
+     **/
+    @Test
+    public void bookClassHasCorrectFields() {
+        String missingFieldMessage = "Your " + bookLc + " class is missing a required field.";
```

Comment:
Do we think that the ` is missing a required field` is detailed enough? I did it because I did not want to give the answer away, but it might be too vague for the student.

Reply:
I think it should be. They should know what fields are.

Reply:
We may want to call it either "attribute" or "field" though, currently we're using both. "Attribute" in the question text and "field" in the error message. Which do you prefer?

Reply:
attribute